lib/net.js

```diff
@@ -357,7 +358,7 @@ Socket.prototype._final = function(cb) {
   req.callback = cb;
   var err = this._handle.shutdown(req);
 
-  if (err === 1)  // synchronous finish
+  if (err === 1 || err === UV_ENOTCONN)  // synchronous finish
```

Comment:
nit: maybe commenting the "unclarity" ?

Reply:
@devnexen Hm yeah, would have been happy to do that in the original PR, but there probably isn’t much point in doing that for the backport

Reply:
ah sorry did not see it as backport.